### PR TITLE
Use absolute target names in poco

### PR DESCRIPTION
**Pull request by tapia**
Conan feature related: https://github.com/conan-io/conan/pull/10099

### CODE REVIEW

File: recipes/poco/all/conanfile.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "Poco")
-        self.cpp_info.set_property("cmake_target_name", "Poco")
+        self.cpp_info.set_property("cmake_target_name", "Poco::Poco")
```
Comment (SpaceIm) [inline recipes/poco/all/conanfile.py:218]: This target doesn't exist upstream.

**Review by czoido — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by lasote — APPROVED** [review]

File: recipes/poco/all/conanfile.py [hunk 2]
```diff
@@ -230,7 +230,8 @@ def package_info(self):
             if comp.option is None or self.options.get_safe(comp.option):
                 conan_component = "poco_{}".format(compname.lower())
                 requires = ["poco_{}".format(dependency.lower()) for dependency in comp.dependencies] + comp.external_dependencies
-                self.cpp_info.components[conan_component].set_property("cmake_target_name", compname)
+                self.cpp_info.components[conan_component].set_property("cmake_target_name", "Poco::{}".format(compname))
+                self.cpp_info.components[conan_component].set_property("cmake_file_name", compname)
```
Comment (SpaceIm) [inline recipes/poco/all/conanfile.py:234]: What is the meaning of `cmake_file_name` for components?

Comment (czoido) [reply recipes/poco/all/conanfile.py:234]: I think that's a mistake, it won't have effect but it should not be set for components.

**Pull request merged**